### ISSUE TITLE
Docker stopped working, restarts won't help

### ISSUE THREAD

**Issue by therepo90** [awslabs/aws-sam-cli#1233]
At one point using sam build --use-container the docker hanged and can't unhang :
```
2019-06-14 10:22:16 Mounting [mypath]/lambda as /tmp/samcli/source:ro,delegated inside runtime container

Build Failed
Error: NodejsNpmBuilder:NpmPack - NPM Failed: npm ERR! code ENOLOCAL
npm ERR! Could not install from "../source" as it does not contain a package.json file.
```

**Comment by jfuss** [comment]
@therepo90 Can you provide more details here? What is your folder structure like? Snippet of template would be useful. 

Are you trying to use local dependencies in your package.json? Version of the cli?

**Comment by TheSriram** [comment]
I'm closing this issue, feel free to re-open it if there are more details to share.

**Comment by MWeesenaar** [comment]
Hello,

I am seeing the exact same message on my build; however I cannot really conclude that it is really sam-related, or it doesn't work given certain circumstances. So my intention of this comment, is just to provide some extra information around my use-case. 

We're running a relatively complex CICD pipeline where I see it: we use Jenkins on AWS ECS, on EC2. From there we execute `sam validate --template x` and `sam build --template x -u`. The build step fails, with the error given by @therepo90. On my Ubuntu 16.04, it does work properly. 
I have tried sam version 0.12.4 and the current latest (0.18.0, from the top of my head??). 

At two days of fiddling around, I found out that if I remove the -u toggle, it works smoothly and I can build and deploy the template. So it must be something Docker related. 

However, where I cannot wrap my head around, is that the build step looks up a package.json which is outside of the source scope. Or, some magic might happen and the package.json that I do provide, is suddenly a folder higher up.

Please let me know if I can be of help with reproducing this issue.

**Comment by orlando1409** [comment]
I found the exact same error in my CI/CD environment, we used Docker inside a CircleCI environment. I tried using the following images:

- lambci/lambda:build-nodejs4.3
- lambci/lambda:nodejs8.10
- lambci/lambda:nodejs10.0

The error is the same. However when I change the run-time to Python (including all the functions) it works:

- lambci/lambda:python3.7

This is most likely a Docker related problem, however I wonder if you guys are working on a way to solve this limitation from your side.

Let me know if can provide more details to reproduce the issue.
